test/integration/viewer_spec.mjs

```diff
@@ -42,23 +42,41 @@ describe("PDF viewer", () => {
       await closePages(pages);
     });
 
-    async function getTextAt(page, pageNumber, coordX, coordY) {
-      await page.waitForFunction(
-        pageNum =>
-          !document.querySelector(
-            `.page[data-page-number="${pageNum}"] > .textLayer`
-          ).hidden,
-        {},
-        pageNumber
+    async function waitForTextAfterZoom(page, originX, originY, scale, text) {
+      const handlePromise = await createPromise(page, resolve => {
+        const callback = e => {
+          if (e.pageNumber === 2) {
+            window.PDFViewerApplication.eventBus.off(
+              "textlayerrendered",
+              callback
+            );
+            resolve();
+          }
+        };
+        window.PDFViewerApplication.eventBus.on("textlayerrendered", callback);
+      });
+
+      await page.evaluate(
+        (scaleFactor, origin) => {
+          window.PDFViewerApplication.pdfViewer.updateScale({
+            drawingDelay: 0,
+            steps: 1,
```

Comment:
This line shouldn't be necessary, since `steps` isn't used when a `scaleFactor` is provided; [please see this code](https://github.com/mozilla/pdf.js/blob/712d502561a29df3378dd7fb7cca7a22e4a48ea9/web/pdf_viewer.js#L2289-L2301).
```suggestion
```